### PR TITLE
Feature/frontend create and delete relationship

### PR DESCRIPTION
**Pull request by manwithtwowatches**
https://www.openproject.org/work_packages/13948
https://www.openproject.org/work_packages/13949

### CODE REVIEW

File: app/assets/javascripts/angular/work_packages/tabs/work-package-relation-directive.js [hunk 1]
```diff
@@ -53,6 +62,23 @@ angular.module('openproject.uiComponents')
         setExpandState();
       });
 
+      // Massive hack alert - Using old prototype autocomplete ///////////
```
Comment (manwithtwowatches) [inline app/assets/javascripts/angular/work_packages/tabs/work-package-relation-directive.js]: @myabc @ulferts Opinions wanted - Is this (or something like this) an acceptable temporary solution for this sprint? For me it doesn't really make sense reimplementing it until we have an api endpoint for work packages index.

Comment (myabc) [inline app/assets/javascripts/angular/work_packages/tabs/work-package-relation-directive.js]: @manwithtwowatches IMO, yes

File: lib/services/create_relation.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      r.relation_type = attrs[:relation_type]
+      r.delay = attrs[:delay]
+    end
+  end
+
+  def run(success = -> {}, failure = -> {})
+    if @relation.save
+      success.(created: true)
+    else
+      binding.pry
```
Comment (hschink) [inline lib/services/create_relation.rb]: eeek

File: public/templates/work_packages/tabs/relations.html [hunk 3]
```diff
@@ -1,58 +1,75 @@
...
+    </work-package-children>
 
-    <work-package-relation title="{{ I18n.t('js.relation_labels.relatedTo') }}"
-                           related-work-packages="relatedTo"
-                           button-title="Add Relation"
+    <work-package-relations title="{{ I18n.t('js.relation_labels.relatedTo') }}"
+                           work-package="workPackage"
+                           relations="relatedTo"
+                           relation-identifier="'relates'"
+                           button-title="Add Related to"
```
Comment (hschink) [inline public/templates/work_packages/tabs/relations.html]: Indentation... see the other ones too.

File: public/templates/work_packages/tabs/relations.html [hunk 4]
```diff
@@ -1,58 +1,75 @@
...
   <div class="detail-panel-description-content">
-    <work-package-relation title="{{ I18n.t('js.relation_labels.parent') }}"
-                           related-work-packages="wpParent"
+    <work-package-parent title="{{ I18n.t('js.relation_labels.parent') }}"
+                           work-package="workPackage"
+                           parent="wpParent"
+                           relation-identifier="'parent'"
                            button-title="Change Parent"
                            button-icon="<i class='icon-hierarchy icon-edit'></i>"
                            singleton-relation="true">
```
Comment (hschink) [inline public/templates/work_packages/tabs/relations.html]: Indentation

File: app/assets/javascripts/angular/work_packages/tabs/related-work-package-table-row-directive.js [hunk 5]
```diff
@@ -27,45 +27,36 @@
...
-          iconClass += '2';
-        }
-
-        return iconClass;
-      }
+      scope.removeRelation = function() {
+        WorkPackageService.removeWorkPackageRelation(scope.relation).then(function(response){
+          scope.$emit('workPackageRefreshRequired', '');
+        }, function(error) {
+          ApiHelper.handleError(scope, error);
```
Comment (hschink) [inline app/assets/javascripts/angular/work_packages/tabs/related-work-package-table-row-directive.js:56]: This is not defined.

File: lib/api/v3/relations/relations_api.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+              model = ::API::V3::WorkPackages::RelationModel.new(relation)
+              representer = ::API::V3::WorkPackages::RelationRepresenter.new(model)
+              relation.save!
+              representer.to_json
+            else
+              raise ::API::Errors::Validation.new(relation)
+            end
+          end
+
+          namespace ':id' do
```
Comment (hschink) [inline lib/api/v3/relations/relations_api.rb]: That won't work in the context of another API that already introduces `:id` namespace (e.g., work packages). Thus, we should rename this parameter to `:relation_id` or something similar.

**Comment by hschink** [comment]
Ready for merge as soon as the tests pass! :+1:

**Pull request merged**